### PR TITLE
feat: flattened json audit format

### PR DESCRIPTION
**Pull request by jptosso**
Following #266, new flattened JSON format.

Friendly with SIEMs, and easy to process, the new struct is:

You can use it like:
```
SecAuditLogFormat flatten
```

```json
{
  "messages.0.actionset": "",
  "messages.0.data.accuracy": 0,
  "messages.0.data.data": "",
  "messages.0.data.file": "",
  "messages.0.data.id": 0,
  "messages.0.data.line": 0,
  "messages.0.data.maturity": 0,
  "messages.0.data.msg": "some message",
  "messages.0.data.rev": "",
  "messages.0.data.severity": 0,
  "messages.0.data.tags": "",
  "messages.0.data.ver": "",
  "messages.0.message": "some message",
  "timestamp": 0,
  "transaction.client_ip": "",
  "transaction.client_port": 0,
  "transaction.host_ip": "",
  "transaction.host_port": 0,
  "transaction.id": "123",
  "transaction.producer.connector": "",
  "transaction.producer.rule_engine": "",
  "transaction.producer.server": "",
  "transaction.producer.stopwatch": "",
  "transaction.producer.version": "",
  "transaction.request.body": "",
  "transaction.request.headers.some": "somedata",
  "transaction.request.http_version": "",
  "transaction.request.method": "GET",
  "transaction.request.uri": "/test.php",
  "transaction.response.headers.some": "somedata",
  "transaction.response.status": 200,
  "transaction.server_id": ""
}
```

### CODE REVIEW

File: types/waf.go [hunk 1]
```diff
@@ -133,7 +133,7 @@ func ParseRequestBodyLimitAction(rbla string) (RequestBodyLimitAction, error) {
 	return -1, fmt.Errorf("invalid request body limit action: %s", rbla)
 }
 
-type auditLogPart byte
+type AuditLogPart byte
```
Comment (fzipi) [inline types/waf.go:136]: Does this classify as API change?

Comment (jptosso) [reply types/waf.go:136]: only if we make it private

**Review by fzipi — APPROVED** [review]
Cool, thanks. LGTM

**Review by piyushroshan — APPROVED** [review]

File: loggers/formats_test.go [hunk 2]
```diff
@@ -72,9 +73,39 @@ func TestLegacyFormatter(t *testing.T) {
 	require.Equal(t, "some message", legacyAl.AuditData.Messages[0], "failed to match legacy formatter")
 }
 
+func TestFlattenFormatter(t *testing.T) {
+	al := createAuditLog(t)
+	bts, err := flattenFormatter(al)
+	if err != nil {
```
Comment (fzipi) [inline loggers/formats_test.go:79]: Now this needs to be changed to assertions, right?

File: loggers/formats_test.go [hunk 3]
```diff
@@ -72,9 +73,39 @@ func TestLegacyFormatter(t *testing.T) {
...
+		t.Error(err)
+	}
```
Comment (fzipi) [inline loggers/formats_test.go:81]:
```suggestion
	assert.NoError(t, err, "failed to create formatter")
```

**Comment by fzipi** [comment]
@jptosso There is a conflict now, looks like.

File: types/waf.go (hunk 1 above)
Comment (jcchavezs) [reply types/waf.go:136]: Could you elaborate a bit on why this becomes public?

**Comment by jptosso** [comment]
It will become a plugin

**Pull request closed without merging**